File: user-resource-service/src/test/java/com/movies/user/util/TestUtils.java
package com.movies.user.util;

import com.movies.common.error.ErrorInfo;
import com.movies.common.error.ErrorType;
import org.springframework.security.oauth2.provider.OAuth2Request;
import org.springframework.test.web.servlet.ResultMatcher;

import java.util.Collections;

import static org.assertj.core.api.Assertions.assertThat;

public class TestUtils {
    public static final String DEFAULT_CLIENT_ID = "testClient";
    public static final OAuth2Request DEFAULT_OAUTH_REQUEST = new OAuth2Request(
            Collections.singletonMap("client_id", DEFAULT_CLIENT_ID),
            DEFAULT_CLIENT_ID, Collections.emptySet(),
            true, Collections.singleton("user_info"),
            Collections.emptySet(), "http://localhost:8080",
            Collections.emptySet(), Collections.emptyMap());

    public static ResultMatcher errorType(ErrorType type, Class<? extends ErrorInfo> responseType) {
        return result -> assertThat(JsonUtil.readFromJson(result, responseType).getType()).isEqualTo(type);
    }
}

File: movies-list-service/src/test/kotlin/com/movies/list/utils/TestUtils.kt
package com.movies.list.utils

import com.movies.common.user.User
import com.movies.common.user.UserRoles
import org.springframework.security.oauth2.provider.OAuth2Request
import java.time.LocalDateTime
import java.util.*

object TestUtils {
    const val DEFAULT_CLIENT_ID = "testClient"
    val DEFAULT_OAUTH_REQUEST = OAuth2Request(
            Collections.singletonMap("client_id", DEFAULT_CLIENT_ID),
            DEFAULT_CLIENT_ID, emptySet(),
            true, setOf("user_info"), emptySet(), "http://localhost:8080", emptySet(), emptyMap())
    val DEFAULT_USER = User(100, "Nikita", "Ivchenko", "munoongg@gmail.com", "{noop}pass", LocalDateTime.now(), setOf(UserRoles.ROLE_USER, UserRoles.ROLE_ADMIN))
}